cpp/src/interop/to_arrow_device.cu

```diff
@@ -586,6 +402,193 @@ int dispatch_to_arrow_device::operator()<cudf::dictionary32>(cudf::column&& colu
+    NANOARROW_RETURN_NOT_OK(set_null_mask(column, tmp.get()));
+    NANOARROW_RETURN_NOT_OK(set_view_to_buffer(column, tmp.get()));
+
+    ArrowArrayMove(tmp.get(), out);
+    return NANOARROW_OK;
+  }
+
+  int set_null_mask(column_view column, ArrowArray* out) const
+  {
+    if (column.nullable()) {
+      NANOARROW_RETURN_NOT_OK(set_buffer_view(column.null_mask(),
+                                              bitmask_allocation_size_bytes(column.size()),
+                                              validity_buffer_idx,
+                                              out));
+    }
+    return NANOARROW_OK;
+  }
+
+  int set_view_to_buffer(column_view column, ArrowArray* out) const
+  {
+    auto const type_size = cudf::size_of(column.type());
+    return set_buffer_view(column.head<uint8_t>() + (type_size * column.offset()),
+                           column.size() * type_size,
+                           fixed_width_data_buffer_idx,
+                           out);
+  }
+};
+
+template <>
+int dispatch_to_arrow_device_view::operator()<numeric::decimal32>(ArrowArray* out) const
+{
+  using DeviceType = int32_t;
+  NANOARROW_RETURN_NOT_OK(decimals_to_arrow<DeviceType>(column, stream, mr, out));
+  NANOARROW_RETURN_NOT_OK(set_null_mask(column, out));
+  return NANOARROW_OK;
+}
+
+template <>
+int dispatch_to_arrow_device_view::operator()<numeric::decimal64>(ArrowArray* out) const
+{
+  using DeviceType = int64_t;
+  NANOARROW_RETURN_NOT_OK(decimals_to_arrow<DeviceType>(column, stream, mr, out));
+  NANOARROW_RETURN_NOT_OK(set_null_mask(column, out));
+  return NANOARROW_OK;
+}
+
+template <>
+int dispatch_to_arrow_device_view::operator()<numeric::decimal128>(ArrowArray* out) const
+{
+  nanoarrow::UniqueArray tmp;
+
+  NANOARROW_RETURN_NOT_OK(initialize_array(tmp.get(), NANOARROW_TYPE_DECIMAL128, column));
+  NANOARROW_RETURN_NOT_OK(set_null_mask(column, tmp.get()));
+  NANOARROW_RETURN_NOT_OK(set_view_to_buffer(column, tmp.get()));
+
+  ArrowArrayMove(tmp.get(), out);
+  return NANOARROW_OK;
+}
+
+template <>
+int dispatch_to_arrow_device_view::operator()<bool>(ArrowArray* out) const
+{
+  nanoarrow::UniqueArray tmp;
+  NANOARROW_RETURN_NOT_OK(initialize_array(tmp.get(), NANOARROW_TYPE_BOOL, column));
+
+  auto bitmask = bools_to_mask(column, stream, mr);
+  NANOARROW_RETURN_NOT_OK(
+    set_buffer(std::move(bitmask.first), fixed_width_data_buffer_idx, tmp.get()));
+  NANOARROW_RETURN_NOT_OK(set_null_mask(column, tmp.get()));
+
+  ArrowArrayMove(tmp.get(), out);
+  return NANOARROW_OK;
+}
+
+template <>
+int dispatch_to_arrow_device_view::operator()<cudf::string_view>(ArrowArray* out) const
+{
+  nanoarrow::UniqueArray tmp;
+  NANOARROW_RETURN_NOT_OK(initialize_array(tmp.get(), NANOARROW_TYPE_STRING, column));
+
+  if (column.size() == 0) {
+    // TODO: Can't we just not set anything here? The arrow spec says that
+    // you can leave the pointer void if the size is 0.
```

Comment:
No longer relevant, see https://github.com/rapidsai/cudf/pull/15047#discussion_r1543164681

Reply:
Updated the comment. Let me know if I should just delete it.